pmm-tests/pmm-2-0-bats-tests/mongodb-tests.bats

```diff
@@ -1,107 +1,77 @@
-#   run sudo pmm-admin add mongodb mymongo1
-#   [ "$status" -eq 0 ]
-#   echo "${lines[0]}" | grep "OK, already"
-#   echo "${lines[1]}" | grep "OK, already"
-# }
-
-@test "run pmm-admin purge mongodb" {
-COUNTER=0
-for i in $(sudo pmm-admin list | grep "mongodb_instance_" | awk '{print $5}' | grep -v '-'|sort -u) ; do
-	let COUNTER=COUNTER+1
-	run sudo pmm-admin purge mongodb:metrics mongodb_instance_${COUNTER}
-  [ "$status" -eq 0 ]
-  echo "${lines[0]}"
-  echo "${lines[0]}" | grep "OK, data purged"
-done
-}
-
-@test "run pmm-admin rm mongodb" {
-COUNTER=0
-for i in $(sudo pmm-admin list | grep "mongodb_instance_" | awk '{print $5}' | grep -v '-'|sort -u) ; do
-	let COUNTER=COUNTER+1
-	run sudo pmm-admin rm mongodb mongodb_instance_${COUNTER}
-  [ "$status" -eq 0 ]
-  echo "${lines[1]}" 
-  echo "${lines[1]}" | grep "OK, removed"
-done
-}
-
-# @test "run pmm-admin rm mongodb named" {
-#   run sudo pmm-admin rm mongodb mymongo1
-#   [ "$status" -eq 0 ]
-#   echo "${lines[0]}" | grep "OK, no system"
-#   echo "${lines[1]}" | grep "OK, removed"
-# }
-
-@test "run pmm-admin add mongodb queries" {
-	COUNTER=0
-  for i in $(sudo pmm-admin list | grep "mongo" | awk '{print $5}' | grep -v '-') ; do
-		let COUNTER=COUNTER+1
-		URI=${i}
-	  run sudo pmm-admin add mongodb:queries --uri ${URI}  mongo_queries_${COUNTER} 
-	  [ "$status" -eq 0 ]
-	  echo "${lines[0]}" | grep "OK, now monitoring"
-  done
-}
-
-@test "run pmm-admin rm mongodb queries" {
-	COUNTER=0
-  for i in $(sudo pmm-admin list | grep "mongo_queries_" | awk '{print $5}' | grep -v '-') ; do
-		let COUNTER=COUNTER+1
-		URI=${i}
-	  run sudo pmm-admin rm mongodb:queries mongo_queries_${COUNTER} 
-	  [ "$status" -eq 0 ]
-	  echo "${lines[0]}" | grep "OK, removed"
-  done
-}
+#!/usr/bin/env bats
+
+## mongodb
+
+@test "run pmm-admin under regular(non-root) user privileges" {
+if [[ $(id -u) -eq 0 ]] ; then
+        skip "Skipping this test, because you are running under root"
+fi
+run pmm-admin
+echo "$output"
+    [ "$status" -eq 0 ]
+    [ "${lines[0]}" = "usage: pmm-admin [<flags>] <command> [<args> ...]" ]
+}
+
+@test "run pmm-admin under root privileges" {
+if [[ $(id -u) -ne 0 ]] ; then
+        skip "Skipping this test, because you are NOT running under root"
+fi
+run sudo pmm-admin
+echo "$output"
+    [ "$status" -eq 0 ]
+    [ "${lines[0]}" = "usage: pmm-admin [<flags>] <command> [<args> ...]" ]
+}
+
+
+@test "run pmm-admin add mongodb based on running instances" {
+	COUNTER=0
+	IFS=$'\n'
+  for i in $(pmm-admin list | grep "MongoDB" | awk -F" " '{print $3}') ; do
+		let COUNTER=COUNTER+1
+		MONGO_IP_PORT=${i}
+		run pmm-admin add mongodb --use-profiler ${MONGO_IP_PORT} mongo_inst_${COUNTER}
+	  [ "$status" -eq 0 ]
+	  echo "${lines[0]}" | grep "MongoDB Service added"
+  done
+}
+
+@test "run pmm-admin add mongodb again based on running instances" {
+	COUNTER=0
+	IFS=$'\n'
+	for i in $(pmm-admin list | grep "MongoDB" | grep "mongo_inst_" | awk -F" " '{print $3}') ; do
```

Comment:
@puneet0191 I guess it should be changed to " pmm-admin list | grep "MongoDB" | grep "mongo" | awk -F" " '{print $3}'"

Reply:
nope this works fine because I am adding instances with that name before this, it is just to reconfirm addition with same name doesn't work

Reply:
oh, sorry, I missed it :)